sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkThriftServerProtocolVersionsSuite.scala

```diff
@@ -362,11 +362,13 @@ class SparkThriftServerProtocolVersionsSuite extends HiveThriftServer2TestBase {
     }
 
     test(s"$version get interval type") {
-      testExecuteStatementWithProtocolVersion(version, "SELECT interval '1' year '2' day") { rs =>
+      testExecuteStatementWithProtocolVersion(
+        version,
+        "SELECT CAST('1 year 2 day' AS INTERVAL) AS icol") { rs =>
```

Comment:
I wasn't able to set the config in the test. So, have to use the Cast expression to get the legacy interval column.